internal/response-consumer/handler.go

```diff
@@ -171,7 +206,82 @@ func inferStatus(events *[]message.PlaybookRunResponseMessageYamlEventsElem, hos
 	}
 }
 
-func getHeaders(msg *k.Message) (requestId string, correlationId uuid.UUID, err error) {
+func inferSatStatus(events *[]message.PlaybookSatRunResponseMessageYamlEventsElem, host *string) string {
+	finished := false
+	failed := false
+	canceled := false
+
+	for _, event := range *events {
+		if event.Type == EventSatPlaybookCompleted {
+			finished = true
+		}
+		if host != nil && event.Host != nil && *event.Host != *host {
+			continue
+		}
+		if event.Type == EventSatPlaybookFinished {
+			finished = true
+		}
+		if event.Status != nil && *event.Status == EventSatStatusCanceled {
+			canceled = true
+		}
+		if event.Status != nil && *event.Status == EventSatStatusFailure {
+			failed = true
```

Comment:
In case of a failure it would be great to make use of the `satellite_connection_error` and `satellite_infrastructure_error` if the error codes are set.

Reply:
I changed this to whenever we see an error code for `satellite_connection_error` or `satellite_infrastructure_error`, copy the error message to console. Added tests for this behavior as well.